src/plugins/discover/public/application/apps/main/components/doc_table/lib/row_formatter.tsx

```diff
@@ -29,18 +30,30 @@ const TemplateComponent = ({ defPairs }: Props) => {
   );
 };
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export const formatRow = (hit: Record<string, any>, indexPattern: IndexPattern) => {
+export const formatRow = (
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  hit: Record<string, any>,
+  indexPattern: IndexPattern,
+  showMultiFields: boolean
+) => {
   const highlights = hit?.highlight ?? {};
   // Keys are sorted in the hits object
   const formatted = indexPattern.formatHit(hit);
   const fields = indexPattern.fields;
   const highlightPairs: Array<[string, unknown]> = [];
   const sourcePairs: Array<[string, unknown]> = [];
+  const fieldNames = fields.map((field) => field.name);
+  const fieldsToShow = getFieldsToShow(fieldNames, indexPattern, showMultiFields);
```

Comment:
This should be optimized for performance, since fields to show is calculated in every row, but it's always equal. As a result there are lots of redundant executions, added a few console.count to show:

<img width="1447" alt="Discover_-_Elastic" src="https://user-images.githubusercontent.com/463851/130462496-d8633fc5-b529-4846-bd6b-7836b2c26e94.png">